stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -214,11 +214,6 @@ function (*)(D::Diagonal, V::AbstractVector)
     return D.diag .* V
 end
 
-(*)(A::AbstractTriangular, D::Diagonal) =
```

Comment:
Wouldn't  this still be needed for custom subtypes of `AbstractTriangular` other than the ones defined in `LinearAlgebra`?

Reply:
We have them in L552, which invoke to the unoptimized version .